packages/cli/src/databases/migrations/common/1717498465932-RefactorExecutionIndices.ts

```diff
@@ -0,0 +1,87 @@
+import type { MigrationContext, ReversibleMigration } from '@/databases/types';
+
+/**
+ * Add new indices:
+ *
+ * - `status, startedAt` for default query at `ExecutionRepository.findManyByRangeQuery`
+ * - `workflowId, status, startedAt` for filtered query at `ExecutionRepository.findManyByRangeQuery`
+ * - `waitTill, status` for regular queries at `ExecutionRepository.getWaitingExecutions`
+ *
+ * Remove unused indices in sqlite:
+ *
+ * - `stoppedAt` (duplicate with different casing)
+ * - `waitTill`
+ * - `status, workflowId`
+ *
+ * Remove unused indices in MySql:
+ *
+ * - `status`
+ *
+ * Remove unused indices in all DBs:
+ *
+ * - `stopped_at`
+ * - `waitTill, id`
+ * - `workflowId, id`
+ *
+ * Keep index as is:
+ *
+ * - `deletedAt` for query at `ExecutionRepository.hardDeleteSoftDeletedExecutions`
+ */
+export class RefactorExecutionIndices1717498465932 implements ReversibleMigration {
+	async up({ schemaBuilder, isPostgres, isSqlite, isMysql }: MigrationContext) {
+		await schemaBuilder.createIndex('execution_entity', ['status', 'startedAt']);
```

Comment:
should we add support for using `CONCURRENTLY` on `CREATE INDEX` on postgres, and use that here to avoid locking up the DB? or does that not make sense since our migrations run before the app start?

Reply:
Today I learned :)

Based the [PG docs](https://www.postgresql.org/docs/current/sql-createindex.html#SQL-CREATEINDEX-CONCURRENTLY) and on my understanding of typeorm migrations, if we use `CONCURRENTLY` we'd avoid blocking inserts, updates and deletes while this migration is running, but there is no such operation in this migration. And I expect inserts, updates and deletes from later migrations would not benefit from `CONCURRENTLY` because migrations are sequential, so later migrations will run only after the current migration has finished.